cognite/client/_api/datapoints.py

```diff
@@ -646,23 +647,24 @@ def insert_dataframe(self, dataframe, external_id_headers: bool = False):
                 >>> c.datapoints.insert_dataframe(df)
         """
         np = utils._auxiliary.local_import("numpy")
-        assert not dataframe.isnull().values.any(), "Dataframe contains NaNs. Remove them in order to insert the data."
-
-        assert np.isfinite(dataframe.select_dtypes(include=[np.number])).values.all(
+        assert not np.isinf(dataframe.select_dtypes(include=[np.number])).any(
             axis=None
         ), "Dataframe contains Infinity. Remove them in order to insert the data."
+        if not dropna:
+            assert (
+                not dataframe.isnull().any()
+            ), "Dataframe contains NaNs. Remove them in order to insert the data."
```

Comment:
This throws `ValueError: The truth value of a Series is ambiguous` currently:
```suggestion
        if not dropna:
            assert (
                not df.isna().values.any()
            ), "Dataframe contains NaNs. Remove them or pass `dropna=True` in order to insert the data."""
```